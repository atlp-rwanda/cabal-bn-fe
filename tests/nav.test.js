/* eslint-disable no-unused-expressions */
import * as React from 'react';
// import "jest-dom/extend-expect"
import renderer from 'react-test-renderer';
import NavBar from '../src/components/navBar';

describe('Navbar', () => {
  it('should render the menubar on click', () => {
    const component = renderer.create(<NavBar />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    renderer.act(() => {
      tree.props.close;
    });

    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
