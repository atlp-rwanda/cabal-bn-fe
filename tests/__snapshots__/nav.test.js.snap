// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render the menubar on click 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1y6ewmh-MuiPaper-root-MuiAppBar-root"
>
  <div
    className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-12bp6ck-MuiToolbar-root"
  >
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-19anirh-MuiTypography-root-MuiLink-root"
      href="/"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <img
        alt="Logo"
        src="Logo.svg"
        style={
          Object {
            "height": 50,
            "width": 100,
          }
        }
      />
    </a>
    <div
      className="MuiBox-root css-9np8v5"
    />
    <div
      className="MuiBox-root css-1m04nb5"
      onClick={[Function]}
    >
      <img
        alt="menu bar"
        src="MenuBar.svg"
      />
    </div>
  </div>
</header>
`;

exports[`Navbar should render the menubar on click 2`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1y6ewmh-MuiPaper-root-MuiAppBar-root"
>
  <div
    className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-12bp6ck-MuiToolbar-root"
  >
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-19anirh-MuiTypography-root-MuiLink-root"
      href="/"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <img
        alt="Logo"
        src="Logo.svg"
        style={
          Object {
            "height": 50,
            "width": 100,
          }
        }
      />
    </a>
    <div
      className="MuiBox-root css-9np8v5"
    />
    <div
      className="MuiBox-root css-1m04nb5"
      onClick={[Function]}
    >
      <img
        alt="menu bar"
        src="MenuBar.svg"
      />
    </div>
  </div>
</header>
`;
